Uses formik hooks for JT Form

AlexSCorey opened the pull request:
##### SUMMARY
This addresses #6317. It converts JobTemplateForm.jsx to using formik hooks.   It also remove handleBlur ( a prop that was passed into the JT Form at some point, but is no longer being used).

##### ISSUE TYPE
\-enhancement

##### COMPONENT NAME
 - UI

##### AWX VERSION
```
9.3
```


##### ADDITIONAL INFORMATION

AlexSCorey committed:
Uses formik hooks for JT Form

jlmitch5 commented on awx/ui_next/src/screens/Template/shared/JobTemplateForm.jsx:
```diff
@@ -62,6 +61,38 @@ function JobTemplateForm({
 
+  const formikBag = useFormikContext();
+  const formikValues = formikBag.values;
+  const setFormikFieldValue = formikBag.setFieldValue;
```
I'd use the setValue off of particularFieldHelpers instead of this.

jlmitch5 commented on awx/ui_next/src/screens/Template/shared/JobTemplateForm.jsx:
```diff
@@ -62,6 +61,38 @@ function JobTemplateForm({
   );
 
+  const formikBag = useFormikContext();
+  const formikValues = formikBag.values;
```
I'd get the value off the first or second (meta) items of the array returned from the useField call

jlmitch5 commented on awx/ui_next/src/screens/Template/shared/JobTemplateForm.jsx:
```diff
@@ -214,108 +234,73 @@ function JobTemplateForm({
+          <PlaybookSelect
+            projectId={formikValues.project}
+            isValid={!(playbookMeta.touched || playbookMeta.error)}
+            form={formikBag}
```
The form prop can be removed here and from PlaybookSelect...it's not actually used in AnsibleSelect

jlmitch5 commented on awx/ui_next/src/screens/Template/shared/JobTemplateForm.jsx:
```diff
@@ -62,6 +61,38 @@ function JobTemplateForm({
     Boolean(template?.host_config_key)
   );
 
+  const formikBag = useFormikContext();
```
If you do the other things mentioned, you won't need to call useFormikContext or have a hold of the formikBag

jlmitch5 commented on awx/ui_next/src/screens/Template/shared/JobTemplateForm.jsx:
```diff
@@ -214,108 +234,73 @@ function JobTemplateForm({
+            touched={inventoryMeta.touched}
+            error={inventoryMeta.error}
+          />
+          {(inventoryMeta.touched || formikValues.ask_inventory_on_launch) &&
```
while this isn't a change you made in this pr, I'm wondering if we can utilize the error system other things use, rather than throwing a custom div here.

(this is also the one place where my plan to excise formikContext might not be feasible...we might not be able to grab just from the useField calls and need the formikContext...not sure off the top of my head

jlmitch5 reviewed (changes requested):
I think these changes are all pretty straightforward, except for the error ask_inventory_on_launch thing...if we need to get together to see if we can figure out a better plan there, I'm happy to help!

AlexSCorey committed:
Removes uncessary formikContext items in favor of useField.
Removed OrgId value from formik and get that value from project field
Updates tests and type.js to reflect those changes.

AlexSCorey commented on awx/ui_next/src/screens/Template/shared/PlaybookSelect.jsx:
```diff
@@ -5,15 +5,7 @@ import { t } from '@lingui/macro';
-  onError,
-  i18n,
-}) {
+function PlaybookSelect({ projectId, isValid, field, onBlur, onError, i18n }) {
```
The only real change here is that `form` was removed.

jlmitch5 reviewed (approved):
changes look good, thanks!

dsesami reviewed (approved):
looks great!

dsesami commented:
recheck
